### DISCUSSION TITLE
Discrepancies in convergence of NACA0012 airfoil between DAFoam and OpenFOAM

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#589] [General]
Hi,

We are testing a NACA0012 airfoil with field inversion. 

It is at Reynolds number 75,000.

I use wall functions so have an average y+ of 100 (created using pyHyp). 

When I run the case using ```mpirun -np 4 simpleFoam -parallel``` i get good convergence (below ```1e-05```).

But if I run the same thing using dafoam (using the ```sample_run.py``` script) and setting the default to ```-runPrimal```, the pressure residuals oscillate at ```7e-04``` and the velocity residuals do not go below ```1.5e-05```.

Attaching the case for you to reproduce - 
[circular_mesh_20K.zip](https://github.com/mdolab/dafoam/files/14313616/circular_mesh_20K.zip)
we are using dafoam v3.0.6 and SNOPT 7.7 as the optimizer. 

Attaching also the pyHyp file used for generating the airfoil grid. 
[genAirFoilMesh.zip](https://github.com/mdolab/dafoam/files/14313614/genAirFoilMesh.zip)

Please let us know what is going on.

**Comment by friedenhe** [comment]
simpleFoam and DASimpleFoam are not exactly the same. Check: https://dafoam.github.io/mydoc_get_started_faq.html#does-dafoam-use-the-exact-same-primal-solvers-in-openfoam  If you want DASimpleFoam to match simpleFoam, set `"useConstrainHbyA": True` in daOption.

**Reply by uttamcadambi07** [reply]
Ah I see, okay this makes sense. Thanks a lot!
